lib/stellar/sep10.rb

```diff
@@ -262,6 +262,8 @@ def self.verify_challenge_tx_threshold(
       signers.each do |s|
         if !signer_strs_found.include?(s['key'])
           next
+        else
+          signer_strs_found.delete(s['key'])
```

Comment:
This is a smart way of getting around the fact that you can't index efficiently into the original `signers` array. 👏 When I was originally looking at this code I didn't see a good way of doing this. This is so simple, love it.